### ISSUE TITLE
Showon removes fields from front end even when they should be shown 

### ISSUE THREAD

**Issue by crystalenka** [JoomTools/plg_system_jtcfshowon#3]
Hi there!

I discovered an issue in Joomla 4 where sometimes, showon removes custom fields from jcfields in the front end view even if they should be shown based on the set conditions.

So for example, I had a field "show announcement" yes/no and only would show the editor field when "show announcement" was set to yes. It would show in the back end form and display the fields correctly, and saved it to the database, but the editor field was not added to jcfields in the front end so I couldn't do anything with it.

It does not seem to happen all the time. I'm not sure how to reproduce it reliably.

I've worked around it now by changing line 214 in `jtcfshowon.php` to:

``` PHP
public function __onContentPrepare($context, &$item, &$params, $page = 0)
```

I'm not sure if that's the best solution (adding the underscores) but it solves the issue.

Happy to help test another solution if you have a better one!

Thank you :)

**Comment by degobbis** [comment]
Hi there!

Thank You for bringing my attention to this.

The method has the task to remove the output of the field by field plugin call `{field ID}` from the content area if it should not be displayed due to the showon rule.

This is done by comparing the name (field->name) of the field from the list of fields not to be displayed (set on line 163 or 181) and searching for the field call and removing it from the content area.

I just added this method to be able to display the fields even more flexible and not only in the `beforeContent` or `afterContent`.

I also cannot explain why this does not always seem to work correctly.

I will keep an eye on this though and do some more tests.

**Comment by crystalenka** [comment]
I was not using it in the content area, I am using template overrides and calling the custom fields directly. Currently, it seems to remove it entirely from the array that jcfields returns. :(

**Comment by crystalenka** [comment]
I have a suspicion. Are you accounting for the NOT operator in the showon field?

Ie, `somefield!:0`? I noticed you didn't document it but it's core Joomla functionality.

**Comment by degobbis** [comment]
Thanks for the tip, I'll check it out right away.

**Comment by degobbis** [comment]
Can You please test this Version:
[plg_system_jtcfshowon-1.0.5-dev.zip](https://github.com/joomtools/plg_system_jtcfshowon/files/9387769/plg_system_jtcfshowon-1.0.5-dev.zip)

**Comment by crystalenka** [comment]
Yes, that seems to do it! Thank you :)

**Comment by degobbis** [comment]
Fixed with version 1.0.5
Thx @crystalenka
